mesonbuild/mesonlib.py

```diff
@@ -574,7 +574,7 @@ def __cmp(self, other, comparator):
         # otherwise, the version with a suffix remaining is greater
         return comparator(len(self._v), len(other._v))
 
-def _version_extract_cmpop(vstr2):
+def _version_extract_cmpop(vstr2: str) -> tuple:
```

Comment:
`typing.Tuple[typing.Callable[[typing.Any, typing.Any], bool], str]`

It would be nice if you could use a `typing.TypeVar` here, [but you can't.](https://github.com/python/mypy/issues/3028)